packages/core/src/xkeys.ts

```diff
@@ -333,18 +350,21 @@ export class XKeys extends EventEmitter {
 
 	/** Initialize the device. This ensures that the essential information from the device about its state has been received. */
 	public async init(): Promise<void> {
-		const pReceivedVersion = new Promise<void>((resolve) => {
-			this.receivedVersionResolve = resolve
-		})
-		const pReceivedGenerateData = new Promise<void>((resolve) => {
-			this.receivedGenerateDataResolve = resolve
-		})
+		if (this.deviceInfo.productId !== 255) {
```

Comment:
A question for you:
What happens if we try to write to the device? Does it crash? Or nothing?

I'm thinking that if it crashes we might want to guard against writing to it in some other methods as well.

Reply:
If I ever try and write to it, I get the following (so the device doesn't crash, but node does):
```
Error in XKeysWatcher TypeError: Cannot write to hid device
    at NodeHIDDevice.write (/home/pi/companion-dev/xkeys-footpedals-fixed/packages/node/dist/node-hid-wrapper.js:19:21)
    at XKeys._write (/home/pi/companion-dev/xkeys-footpedals-fixed/packages/core/dist/xkeys.js:652:25)
    at XKeys._getVersion (/home/pi/companion-dev/xkeys-footpedals-fixed/packages/core/dist/xkeys.js:701:14)
    at XKeys.init (/home/pi/companion-dev/xkeys-footpedals-fixed/packages/core/dist/xkeys.js:319:18)
    at setupXkeysPanel (/home/pi/companion-dev/xkeys-footpedals-fixed/packages/node/dist/methods.js:77:17)
    at XKeysWatcher.handleNewDevice (/home/pi/companion-dev/xkeys-footpedals-fixed/packages/node/dist/watcher.js:196:32)
    at XKeysWatcher.updateConnectedDevices (/home/pi/companion-dev/xkeys-footpedals-fixed/packages/node/dist/watcher.js:181:22)
    at Timeout._onTimeout (/home/pi/companion-dev/xkeys-footpedals-fixed/packages/node/dist/watcher.js:136:22)
    at listOnTimeout (node:internal/timers:569:17)
    at process.processTimers (node:internal/timers:512:7)
```

So I did the protection in each place that calls write, as it often needed other steps to work around it, but maybe it should be done more centrally?

Reply:
I have been away, but earlier you asked if I could assist. Not sure if I can, the VEC foot pedal is a proprietary device made specifically for VEC and their software. It is not an X-keys so not sure if it belongs here. It is very limited in its capabilities, however, 3 bits is just 3 bits, so it does that. As you are finding there will be a lot of work arounds to make it work.

Reply:
> I have been away, but earlier you asked if I could assist. Not sure if I can, the VEC foot pedal is a proprietary device made specifically for VEC and their software.

Fair enough, I'd found lots of other hits on the Internet, and it's just integrated into Linux via the Event stuff, so I didn't realise it was originally designed just for VEC's software.

> It is not an X-keys so not sure if it belongs here. It is very limited in its capabilities, however, 3 bits is just 3 bits, so it does that. As you are finding there will be a lot of work arounds to make it work.

That seems quite similar with the RailDriver ( https://xkeys.com/xkeys/rdproducts.html ) stuff too, looking at some of the workarounds for that in the code too.

I would say conversely, they're all P.I. Engineering devices, the VID confirms that, and the fact there's an official XKeys device which looks like later revisions of the VEC Infinity range it would seem a bit of a shame to me if the support wasn't added:
https://xkeys.com/xkeys/switches-pedals/footpedals.html
http://veccorp.com/foot-controls.html